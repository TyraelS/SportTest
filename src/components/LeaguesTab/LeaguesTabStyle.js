import styled from 'styled-components';

const LeaguesTabStyle = styled.div`
  width: 300px;
  height: auto;
  display: flex;
  flex-direction: column;
`;

export default LeaguesTabStyle;
