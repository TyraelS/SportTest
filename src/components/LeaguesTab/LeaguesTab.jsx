import React from 'react';

import LeaguesTabStyle from './LeaguesTabStyle';
import LeaguesTabItem from 'Components/LeaguesTabItem';

const displayName = 'LeaguesTab';

const LeaguesTab = ({ leagues }) => {
  const items = leagues.toJS();
  return (
    <LeaguesTabStyle>
      {Object.keys(items).map(key => (
        <LeaguesTabItem
          key={items[key].id}
          name={items[key].name}
          country={items[key].countryCode}
        >
          {items[key].name}
        </LeaguesTabItem>
      ))}
    </LeaguesTabStyle>
  );
};

LeaguesTab.displayName = displayName;

export default LeaguesTab;
