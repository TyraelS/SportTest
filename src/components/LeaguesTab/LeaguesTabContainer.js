import { compose, setDisplayName } from 'recompose';
import { Map } from 'immutable';
import { connect } from 'react-redux';

import LeaguesTab from './LeaguesTab';

export const mapStateToProps = state => ({
  leagues: state.get('leagues', Map())
});

export const enhance = compose(
  setDisplayName('LeaguesTabContainer'),
  connect(
    mapStateToProps,
    null
  )
);

export default enhance(LeaguesTab);
