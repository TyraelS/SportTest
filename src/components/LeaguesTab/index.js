import LeaguesTabContainer from './LeaguesTabContainer';

export default LeaguesTabContainer;
