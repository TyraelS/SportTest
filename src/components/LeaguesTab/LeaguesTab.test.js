import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import LeaguesTab from './LeaguesTab';

describe('Given the LeaguesTab component with provided props', () => {
  const leagues = fromJS({
    SBTC_01: {
      key: 'SBTC_01',
      name: 'SortofLeague',
      countryCode: 'GBR'
    },
    SBTC_02: {
      key: 'SBTC_02',
      name: 'AnotherLeague',
      countryCode: 'USA'
    }
  });
  describe('when the component is rendered', () => {
    const component = shallow(<LeaguesTab leagues={leagues} />);

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
