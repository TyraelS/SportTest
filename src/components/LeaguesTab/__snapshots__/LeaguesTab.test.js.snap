// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the LeaguesTab component with provided props when the component is rendered should match the snapshot 1`] = `
<ForwardRef(styled.div)>
  <LeaguesTabItem
    country="GBR"
    key="SBTC_01"
    name="SortofLeague"
  >
    SortofLeague
  </LeaguesTabItem>
  <LeaguesTabItem
    country="USA"
    key="SBTC_02"
    name="AnotherLeague"
  >
    AnotherLeague
  </LeaguesTabItem>
</ForwardRef(styled.div)>
`;
